lua/core/menu.lua

```diff
@@ -1145,7 +1146,104 @@ m.deinit[pRESET] = function() end
 
 
 -----------------------------------------
+-- UPDATE
+m.update = {}
+m.update.url = ''
+m.update.version = ''
+
+local function check_newest()
+  print("checking for update")
+  m.update.url = util.os_capture( [[curl -s \
+      https://api.github.com/repos/monome/norns/releases/latest \
+      | grep "browser_download_url.*" \
+      | cut -d : -f 2,3 \
+      | tr -d \"]])
+  print(m.update.url)
+  m.update.version = m.update.url:match("(%d%d%d%d%d%d)")
+  print("available version "..m.update.version)
+end
+
+local function get_update()
+  m.update.message = "preparing..."
+  menu.redraw()
+  pcall(cleanup) -- shut down script
+  norns.script.clear()
+  os.execute("sudo systemctl stop norns-jack.service") -- disable audio
+  os.execute("sudo rm -rf /home/we/update/*") -- clear old updates
+  m.update.message = "downloading..."
+  menu.redraw()
+  os.execute("wget -q -P /home/we/update/ " .. m.update.url) --download
```

Comment:
what happens if `wget` fails?

Reply:
i'm not sure what should happen.

i figure "failure" means an indefinite hold... ie, the download has stalled. this process won't even _start_ unless the file was found, and github is generally reliable. if github was down, it wouldn't find the update at all... i suppose this error needs to be caught (which is different than the current error detection which is no nameserver)

honestly i don't even know how to set a time limit to `wget`. is there a way? @simonvanderveldt  @artfwo 

then the following step should be check if the file is there? how do we know if it is complete?

Reply:
basically `wget` defaults to aborting after 900 seconds (15 minutes) of inactivity. i should probably set this to something like 3 minutes. (180s).

unfortunately i can't figure out what `wget` does with a partial file that has quit mid-way. does it delete the file? or leave the partial?

this is a sort of arg-problem, figuring out how to deal with github suddenly going down after a download was initiated. or, someone pulls out their wifi dongle. or kills their local network.

i guess i'll try pulling my wifi. good idea.

Reply:
results from pulling wifi after 10 seconds of downloading:

file is there, but 0 bytes.

i have no idea if this is reproducible behavior across other fail states.

perhaps this suggests that that enclosed .tgz and .md5 are actually a good idea.

Reply:
other option: just check tar output for "unexpected end of file"

Reply:
probably not a bad idea to check for a bogus archive (md5 or otherwise).  maybe for 2.1?

Reply:
We should compare the file we download vs a hash (sha256 or sth like that). If it doesn't match we can either error out or try again. This will cover both half/incorrect downloads as well as other causes for problems (broken upload for example).

Reply:
this is a good idea! i'll do it this way. solves many problems at once.

Reply:
added sha256 hash!!